src/App.jsx

```diff
@@ -1,24 +1,58 @@
-import React, { useRef } from "react";
+import React, { useState, useEffect, useRef } from "react";
 import TodoList from "./TodoList.jsx";
 import AddTodoForm from "./AddTodoForm.jsx";
-import useSemiPersistentState from "./hooks/useSemiPersistentState.jsx";
+//import useSemiPersistentState from "./hooks/useSemiPersistentState.jsx";
 
 const App = () => {
+  const initialTodoList = []; // default empty list
+  const [todoList, setTodoList] = useState([]); // initialize todolist state
+  //using empty array
+  //   () => {
+  //   const savedTodoList = localStorage.getItem("savedTodoList");
+  //   if (savedTodoList) {
+  //     return JSON.parse(savedTodoList);
+  //   } else {
+  //     return initialTodoList;
+  //   }
+  // });
+  const [isLoading, setIsLoading] = useState(true); //new state isLoading
+  const inputRef = useRef(null); //ref for the input element
 
-  const [todoList, setTodoList] = useSemiPersistentState("savedTodoList", []);
-  //ref for input element
-  const inputRef = useRef(null);//ref for the input element
+  useEffect(() => {
+    const myPromise = new Promise((resolve) => {
+      setTimeout(() => {
+        resolve({ data: { todoList: initialTodoList } });
+      }, 2000);
+    });
+
+    myPromise
+      .then((result) => {
+        setTodoList(result.data.todoList);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }, []); //empty dep array runs only once after initial render
+
+  useEffect(() => {
+    // Checking if loading is complete before setting localStorage
+    if (!isLoading) {
+      localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+    }
+  }, [todoList, isLoading]); // Dependency array includes todoList and isLoading
 
   const addTodo = (newTodo) => {
-    setTodoList((prevList) => [...prevList, newTodo]);
+    setTodoList((prevList) => [...prevList, newTodo]); //add new todo
```

Comment:
You need to write the inputRef.current.focus() here, that way you will have your input focused after adding a todo as well.